src/Datagrid/DatagridMapper.php

```diff
@@ -78,41 +76,32 @@ public function getAdmin(): AdminInterface
      * @phpstan-param class-string|null $type
      */
     public function add(
-        $name,
+        string $name,
         ?string $type = null,
         array $filterOptions = [],
         array $fieldDescriptionOptions = []
     ): self {
-        if ($name instanceof FieldDescriptionInterface) {
-            $fieldDescription = $name;
-            $fieldDescription->mergeOptions($filterOptions);
-        } elseif (\is_string($name)) {
-            if ($this->getAdmin()->hasFilterFieldDescription($name)) {
-                throw new \LogicException(sprintf(
-                    'Duplicate field name "%s" in datagrid mapper. Names should be unique.',
-                    $name
-                ));
-            }
-
-            $fieldDescription = $this->getAdmin()->createFieldDescription(
-                $name,
-                array_merge($filterOptions, $fieldDescriptionOptions)
-            );
-        } else {
-            throw new \TypeError(
-                'Unknown field name in datagrid mapper.'
-                .' Field name should be either of FieldDescriptionInterface interface or string.'
-            );
+        if (isset($fieldDescriptionOptions['role']) && !$this->getAdmin()->isGranted($fieldDescriptionOptions['role'])) {
```

Comment:
In FormMapper this was an early check, I think it's a good idea to have an early return instead of doing all the work and not adding the field.